### DISCUSSION TITLE
Has anyone successfully built EdgeDB for ARM?

### DISCUSSION THREAD

**Discussion by spro** [geldata/gel#1503] [Q&A]
I've only found a few mentions around the web...

* https://gist.github.com/fmoor/e54987780622f3f9b9bf878ef6af2d83 - promising title, failed outcome
* https://github.com/edgedb/edgedb-docker/issues/4 - "we lack the necessary infrastructure" but plan to "allow (cross-) compiling to a specific target"

I made an attempt at compiling from source but got stuck at an [unsupported architecture warning](https://github.com/edgedb/edgedb-cli/blob/f206b70aadab5c853ce450150f942a7b99c160e7/src/server/detect.rs#L19). This isn't something I understand well enough to progress from there.

Also tried with Docker and learned that's not the problem Docker solves.

My use case is just a toy project on a Raspberry Pi, but with ARM Macs announced I'm sure it has become relevant to many others...

**Comment by elprans** [comment]
The docker buildx approach by @fmoor might actually be viable. I'll see if I can find some time to play with it to produce a working image.

Regarding the "unsupported architecture" error -- that comes from edgedb-cli, you may be able to progress further by not building it (comment out relevant parts in `class develop` in `setup.py`).

**Reply by unmade** [reply]
I've commented `_compile_cli(build_base, build_temp)` in `class develop` and compiled the EdgeDB from master. 

Now, I'm able to run the server with `edb server`. However, I can't figure out how to connect to the server. I'm using edgedb-cli installed via install-script. The command `edgedb -H localhost` results in error:

```
[2021-08-16T13:46:36Z WARN  rustls::session] Sending fatal alert BadCertificate
edgedb error: invalid certificate: UnknownIssuer
```

~~Any ideas how to fix that?~~

**UPDATE**

Turned out there is a `--tls-ca-file` option

---

By the way, do you have any updates on the docker image for arm architectures?

---
I'm on the MacBook with M1 chip.

**Reply by 1st1** [reply]
In dev build you can run `edb cli` to connect to the instance run by `edb server`

**Reply by unmade** [reply]
I excluded `cli` from build, because it fails with `Unsupported architecture, supported: x86_64`. Because of that I can't use `edb cli`.

**Reply by 1st1** [reply]
@tailhook @elprans any thoughts here?

**Reply by tailhook** [reply]
Ah, sorry, this one is easily fixed: https://github.com/edgedb/edgedb-cli/pull/480
Although, I've not tested it on M1 yet.

**Reply by unmade** [reply]
I've tried to compile it with fix, however I still has this error:

```
error: Unsupported architecture, supported: x86_64, aarm64
      --> src/server/detect.rs:23:1
       |
    23 | compile_error!("Unsupported architecture, supported: x86_64, aarm64");
       | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

    error[E0432]: unresolved import `crate::server::detect::ARCH`
      --> src/server/debian_like.rs:11:35
       |
    11 | use crate::server::detect::{Lazy, ARCH};
       |                                   ^^^^ no `ARCH` in `server::detect`

    error[E0432]: unresolved import `crate::server::detect::ARCH`
      --> src/server/centos.rs:11:35
       |
    11 | use crate::server::detect::{Lazy, ARCH};
       |                                   ^^^^ no `ARCH` in `server::detect`

    error[E0432]: unresolved import `crate::server::detect::ARCH`
      --> src/server/macos.rs:21:29
       |
    21 | use crate::server::detect::{ARCH, Lazy, VersionQuery};
       |                             ^^^^ no `ARCH` in `server::detect`

    error: aborting due to 4 previous errors

    For more information about this error, try `rustc --explain E0432`.
    error: failed to compile `edgedb-cli v1.0.0-beta.3 (https://github.com/edgedb/edgedb-cli?rev=a48441333e9e4b199447e787ed3f48116f2d6e8e#a4844133)`
```

**Reply by tailhook** [reply]
@unmade, yes sorry, there was a typo in the arch name. I've pushed the correct version into the master.

**Reply by unmade** [reply]
It's all good now! I've successfully compiled it and now can use `edb cli`. Once again, thank you so much for taking your time!
